src/index.ts

```diff
@@ -0,0 +1,229 @@
+};
+
+export default class SimpleKeyring implements Keyring<string[]> {
+  #wallets: { privateKey: Buffer; publicKey: Buffer }[];
+
+  public type: string = _TYPE;
+
+  static type: string = _TYPE;
+
+  constructor(options: string[] = []) {
+    this.#wallets = [];
+    this.deserialize(options);
+  }
+
+  async serialize() {
+    return this.#wallets.map((a) => a.privateKey.toString('hex'));
+  }
+
+  async deserialize(privateKeys: string[]) {
+    this.#wallets = privateKeys.map((hexPrivateKey) => {
+      const strippedHexPrivateKey = stripHexPrefix(hexPrivateKey);
+      const privateKey = Buffer.from(strippedHexPrivateKey, 'hex');
+      const publicKey = privateToPublic(privateKey);
+      return { privateKey, publicKey };
+    });
+  }
+
+  async addAccounts(numAccounts = 1) {
+    const newWallets = [];
+    for (let i = 0; i < numAccounts; i++) {
+      const privateKey = this.#generateKey();
+      const publicKey = privateToPublic(privateKey);
+      newWallets.push({ privateKey, publicKey });
+    }
+    this.#wallets = this.#wallets.concat(newWallets);
+    const hexWallets = newWallets.map(({ publicKey }) =>
+      add0x(bufferToHex(publicToAddress(publicKey))),
+    );
+    return hexWallets;
+  }
+
+  async getAccounts() {
+    return this.#wallets.map(({ publicKey }) =>
+      add0x(bufferToHex(publicToAddress(publicKey))),
+    );
+  }
+
+  async signTransaction(
+    address: Hex,
+    transaction: TypedTransaction,
+    options: Record<string, unknown> = {},
+  ): Promise<TxData> {
+    const privKey = this.#getPrivateKeyFor(address, options);
+    const tx = transaction as unknown as Transaction;
+    const signedTx = tx.sign(privKey);
+    // Newer versions of Ethereumjs-tx are immutable and return a new tx object
+    return signedTx === undefined ? tx : signedTx;
+  }
+
+  // For eth_sign, we need to sign arbitrary data:
+  async signMessage(
+    address: Hex,
+    data: string,
+    opts = { withAppKeyOrigin: '' },
+  ) {
+    const message = stripHexPrefix(data);
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    const msgSig = ecsign(Buffer.from(message, 'hex'), privKey);
+    const rawMsgSig = concatSig(toBuffer(msgSig.v), msgSig.r, msgSig.s);
+    return rawMsgSig;
+  }
+
+  // For personal_sign, we need to prefix the message:
+  async signPersonalMessage(
+    address: Hex,
+    msgHex: Hex,
+    opts = { withAppKeyOrigin: '' },
+  ) {
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    // const privateKey = Buffer.from(privKey, 'hex');
+    return personalSign({ privateKey: privKey, data: msgHex });
+  }
+
+  // For eth_decryptMessage:
+  async decryptMessage(withAccount: Hex, encryptedData: Eip1024EncryptedData) {
+    const wallet = this.#getWalletForAccount(withAccount);
+    const privateKey = wallet.privateKey.toString('hex');
+    return decrypt({ privateKey, encryptedData });
+  }
+
+  // personal_signTypedData, signs data along with the schema
+  async signTypedData(
```

Comment:
This could have been much stricter with this signature, avoiding the `any` type on typedData:

```typescript
async signTypedData<V extends SignTypedDataVersion, T extends MessageTypes>(
  address: Hex,
  typedData: V extends 'V1' ? TypedDataV1 : TypedMessage<T>,
  opts: Record<string, string> = { version: SignTypedDataVersion.V1 },
)
```

But as you are implementing the `Keyring` interface this is not possible. Oh well, I guess this goes out of scope for this PR